x-pack/plugins/lens/public/xy_visualization/xy_config_panel.tsx

```diff
@@ -529,22 +565,36 @@ export function DimensionEditor(props: VisualizationDimensionEditorProps<State>)
           idSelected={`${idPrefix}${axisMode}`}
           onChange={(id) => {
             const newMode = id.replace(idPrefix, '') as YAxisMode;
-            const newYAxisConfigs = [...(layer.yConfig || [])];
-            const existingIndex = newYAxisConfigs.findIndex(
-              (yAxisConfig) => yAxisConfig.forAccessor === accessor
-            );
-            if (existingIndex !== -1) {
-              newYAxisConfigs[existingIndex].axisMode = newMode;
-            } else {
-              newYAxisConfigs.push({
-                forAccessor: accessor,
-                axisMode: newMode,
-              });
-            }
+            const newYAxisConfigs = createNewYAxisConfigsWithValue('axisMode', newMode);
             setState(updateLayer(state, { ...layer, yConfig: newYAxisConfigs }, index));
           }}
         />
       </EuiFormRow>
+      {shouldShowValueLabels && (
```

Comment:
Overall I was expecting the value label options to behave more like how we've implemented fitting functions: the form always appears in the same place, but the options are disabled and ignored when they can't be used.